enterprise/api/v1/views/enterprise_group.py

```diff
@@ -42,17 +48,44 @@ def get_queryset(self, **kwargs):
                 associated_customers.append(user_object.enterprise_customer)
             queryset = queryset.filter(enterprise_customer__in=associated_customers)
 
+            if self.request.method in ('PATCH',):
+                if requested_customer := self.request.data.get('enterprise_customer'):
+                    implicit_access = rules.has_implicit_access_to_dashboard(self.request.user, requested_customer)
+                    explicit_access = rules.has_explicit_access_to_dashboard(self.request.user, requested_customer)
+                    if not implicit_access and not explicit_access:
+                        raise Http404
```

Comment:
thinking about it, I chose 404 original because the user shouldn't have the knowledge that the resource exists so pretend like it doesn't, but 401 should be fine in this case